src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -705,6 +718,8 @@ protected internal virtual void AddResponseFileCommands(CommandLineBuilderExtens
             commandLine.AppendPlusOrMinusSwitch("/optimize", _store, nameof(Optimize));
             commandLine.AppendSwitchIfNotNull("/pathmap:", PathMap);
             commandLine.AppendSwitchIfNotNull("/out:", OutputAssembly);
+            commandLine.AppendSwitchIfNotNull("/refout:", OutputRefAssembly);
+            commandLine.AppendWhenTrue("/refonly", _store, nameof(RefOnly));
```

Comment:
Do you intend this to be user-settable in projects, for example if I have some project with manual type definitions so I can make a reference assembly that way? If so, then you should consider moving this to `AddResponseFileCommandsForSwitchesSinceInitialReleaseThatAreNeededByTheHost`.